### DISCUSSION TITLE
How do I find the initial elastic strain guess?

### DISCUSSION THREAD

**Discussion by Moyiqi** [idaholab/moose#24181] [Q&A General] [answered]
Hello everyone,may I ask a question?
![image](https://user-images.githubusercontent.com/59696390/234532795-fc075b2b-6086-47ec-9d5a-7d8bad77c3c4.png)
When I calculated it in the process above, how do I find the initial elastic strain guess 
![image](https://user-images.githubusercontent.com/59696390/234533065-ac3a406b-ec1c-46d3-be96-f35a0379ae3a.png)
?
Lily

**Comment by hugary1995** [comment]
You don't. We do that for you in the material update.

**Reply by Moyiqi** [reply]
Thanks for your reply, does the system have a default value. But I can't adjust it?
Lily

**Reply by hugary1995** [reply, marked as answer]
The initial trial elastic strain is set to be equal to the entirety of the mechanical strain (excluding eigenstrain contributions). I am not aware of any other convergent algorithm which uses a different initial guess. However, there is nothing preventing you from making your own stress calculator, in which you can try out different algorithms.
